### PR TITLE
Enhancement: Replace <img> tags with Next.js Image Component #824

### PR DESCRIPTION
**Pull request by 29deepanshutyagi**
This PR refactors several img tags throughout the project to utilize the Image component provided by Next.js.Replaced img tags with Image components 

changes are pushed through check branch

issue #824 , kindly check it

### CODE REVIEW

**Review by benjagm — CHANGES_REQUESTED** [review]
Thanks a lot for this PR!! Nice job. I just left small comments.

File: components/GettingStarted.tsx [hunk 1]
```diff
@@ -134,7 +135,7 @@ const GettingStarted = () => {
               </label>
               <select
                 name='Select a JSON Schema Validator'
-                className='p-2 border dark:border-slate-300 border-slate-800 dark:bg-slate-900 rounded-md max-sm:text-[12px]'
+                className='p-2 border dark:border-slate-300 border-slate-800 dark:bg-slate-900 rounded-md max-sm:text-[12px] plausible-event-name==activation-explore-tools'
```
[the remark this answers is not in the document]
Comment (29deepanshutyagi) [reply components/GettingStarted.tsx:138]: Sure

**Comment by 29deepanshutyagi** [comment]
@benjagm  all requested changes are done ,kindly check

File: pages/tools/index.page.tsx
Comment (DarhkVoyd) [inline pages/tools/index.page.tsx:1]: These changes break the CTA styles.

**Review by DarhkVoyd — CHANGES_REQUESTED** [review]

File: pages/index.page.tsx
Comment (DarhkVoyd) [inline pages/index.page.tsx:1]: Some of these changes introduces hydration errors on the homepage, kindly check the dev console.

**Review by DarhkVoyd — CHANGES_REQUESTED** [review]

File: pages/tools/index.page.tsx
Comment (29deepanshutyagi) [reply pages/tools/index.page.tsx:1]: can you kindly ellaborate,if possible

File: components/Card.tsx [hunk 2]
```diff
@@ -36,15 +37,24 @@ const CardBody = ({
     <div className='group relative h-full w-full rounded-lg border border-gray-200 bg-white p-6 px-12 shadow-3xl dark:shadow-2xl dark:shadow-slate-900 transition-colors ease-in-out hover:bg-slate-100 dark:bg-slate-800 hover:dark:bg-slate-900/30'>
       <div className='flex justify-center '>
         {image && (
-          <img src={image} className='h-32 p-2' data-test='card-image' />
+          <Image
+            src={image}
+            alt={title}
+            width={128} // Corresponds to h-32 (128px)
```
Comment (DhairyaMajmudar) [inline components/Card.tsx:43]: please remove this extra commnets

File: components/Card.tsx [hunk 3]
```diff
@@ -36,15 +37,24 @@ const CardBody = ({
...
       </div>
       <div className='flex flex-row items-start mb-6'>
         {icon && (
           <span className='mr-6 flex h-14 w-14 flex-shrink-0 items-center justify-center rounded-lg border bg-blue-200 px-3 text-gray-900 dark:text-white'>
-            <img
+            <Image
               src={icon}
               alt={title}
+              width={56} // Corresponds to w-14 (56px)
+              height={56} // Corresponds to h-14 (56px)
```
Comment (DhairyaMajmudar) [inline components/Card.tsx:57]: Please remove extra comments

File: components/GettingStarted.tsx [hunk 4]
```diff
@@ -237,9 +238,19 @@ const GettingStarted = () => {
...
+                height={24} // Height of the icon
+              />
             ) : (
-              <img src='/icons/red-cross.svg' alt='red cross' />
+              <Image
+                src='/icons/red-cross.svg'
+                alt='red cross'
+                width={24} // Width of the icon
+                height={24} // Height of the icon
+              />
```
Comment (DhairyaMajmudar) [inline components/GettingStarted.tsx:253]: Pls. remove extra comments

File: components/Layout.tsx [hunk 5]
```diff
@@ -315,7 +316,13 @@ const Footer = () => (
   >
     <div className='max-w-[1400px] mx-auto mt-8 md:mt-4 grid grid-cols-1 md:grid-cols-2 md:w-1/2 lg:w-1/3 justify-center '>
       <div className=' my-6 m-auto md:mt-16'>
-        <img src='/img/logos/logo-white.svg' className='w-[150px] mb-6' />
+        <Image
+          src='/img/logos/logo-white.svg'
+          width={150}
+          height={100}
+          alt={''}
```
Comment (DhairyaMajmudar) [inline components/Layout.tsx:323]: Fill in the proper alt title

File: components/StyledMarkdown.tsx [hunk 6]
```diff
@@ -378,7 +378,13 @@ const StyledMarkdownBlock = ({ markdown }: { markdown: string }) => {
               component: ({ label }) => {
                 return (
                   <div className='flex flex-row items-center text-blue-500 text-lg font-semibold mb-6 mt-10'>
-                    <img src='/icons/star.svg' className='h-5 w-5 mr-2 mb-1' />
+                    <Image
+                      src='/icons/star.svg'
+                      className='h-5 w-5 mr-2 mb-1'
+                      width={20}
+                      height={20}
+                      alt={''}
```
Comment (DhairyaMajmudar) [inline components/StyledMarkdown.tsx:386]: Fill in  the proper alt title

Comment (29deepanshutyagi) [reply components/StyledMarkdown.tsx:386]: sure

File: pages/blog/posts/[slug].page.tsx [hunk 7]
```diff
@@ -60,7 +61,13 @@ export default function StaticMarkdownPage({
...
                   className='font-semibold text-sm pb-0 lg:pb-5 text-slate-700 dark:text-slate-300 dark:hover:text-slate-100 hover:text-slate-800 inline-flex flex-row items-center'
                 >
-                  <img src='/icons/left-arrow.svg' className='h-4 w-4 mr-2' />{' '}
+                  <Image
+                    src='/icons/left-arrow.svg'
+                    width={16} // Corresponds to 'w-4' in Tailwind (4 * 4 = 16px)
+                    height={16} // Corresponds to 'h-4' in Tailwind
+                    className='h-4 w-4 mr-2'
+                    alt='Left arrow icon' // Add meaningful alt text for accessibility
+                  />
```
Comment (DhairyaMajmudar) [inline pages/blog/posts/[slug].page.tsx:70]:
```suggestion
                    width={16} // Corresponds to 'w-4' in Tailwind (4 * 4 = 16px)
                    height={16}
                    className='h-4 w-4 mr-2'
                    alt='Left arrow icon'
                  />
```

File: pages/community/index.page.tsx [hunk 8]
```diff
@@ -379,9 +379,12 @@ export default function communityPages(props: any) {
             <div className='p-10 flex justify-between w-full md:w-3/6 h-auto flex-col text-center md:text-left '>
               <div className='w-full mb-6 '>
                 <Link href={`/blog/posts/${blogPosts[0].slug}`}>
-                  <img
+                  <Image
                     src={blogPosts[0].frontmatter.cover}
                     className='w-full h-[232px]  mb-4'
+                    height={232}
+                    width={400}
+                    alt={''}
```
Comment (DhairyaMajmudar) [inline pages/community/index.page.tsx:387]: Add meaningful alt text here

File: pages/index.page.tsx [hunk 9]
```diff
@@ -273,22 +280,38 @@ const Home = (props: any) => {
...
-                />
-                <img
-                  src='/img/logos/usedby/postman-white.png'
-                  className='w-40 mx-auto'
-                />
-                <img
-                  src='/img/logos/usedby/github-white.png'
-                  className='w-40 mx-auto'
-                />
+                {isClient && (
```
Comment (DhairyaMajmudar) [inline pages/index.page.tsx:283]: Please remove the conditional rendering of the image and add meaningful alt texts

File: pages/index.page.tsx [hunk 10]
```diff
@@ -377,10 +400,17 @@ const Home = (props: any) => {
...
+            <>
+              <Image
+                src='/img/home-page/community-illustration.svg'
+                className='w-5/6 mx-auto lg:w-[600px] xl:w-[800px]'
+                alt={''}
+                height={600}
+                width={800}
+              />
+            </>
+          )}
```
Comment (DhairyaMajmudar) [inline pages/index.page.tsx:413]: Remove conditional rendering of image and add meaningful alt text

File: pages/index.page.tsx [hunk 11]
```diff
@@ -211,6 +211,13 @@ const Home = (props: any) => {
   const [common_room_logo, setCommon_room_logo] = useState('');
   const [slack_logo, setSlack_logo] = useState('');
 
+  const [isClient, setIsClient] = useState(false);
+
+  useEffect(() => {
+    // Ensure the component is only rendered client-side
+    setIsClient(true);
+  }, []);
+
```
Comment (DhairyaMajmudar) [inline pages/index.page.tsx:220]: No need of this, pls. remove it and conditionally rendered images

File: pages/tools/components/ToolingDetailModal.tsx [hunk 12]
```diff
@@ -42,9 +43,12 @@ export default function ToolingDetailModal({
         <div className='mt-4 flex flex-row items-center justify-start gap-2'>
           {tool.landscape?.logo && (
             <div className='p-2 flex flex-row items-center dark:bg-white rounded-md flex-none'>
-              <img
+              <Image
                 src={`img/tools/logos/${tool.landscape?.logo}`}
                 className='h-[48px] w-[48px]'
+                height={48}
+                width={48}
+                alt={''}
```
Comment (DhairyaMajmudar) [inline pages/tools/components/ToolingDetailModal.tsx:51]:
```suggestion
                alt='landscape logos'
```

File: pages/tools/index.page.tsx [hunk 13]
```diff
@@ -181,9 +182,12 @@ export default function ToolingPage({
...
                   rel='noreferrer'
                 >
-                  <img
+                  <Image
                     src='/img/tools/adding_your_tool.png'
-                    className='rounded-sm h-[68px]'
+                    className='rounded-sm '
+                    height={68}
+                    width={100}
+                    alt={''}
```
Comment (DhairyaMajmudar) [inline pages/tools/index.page.tsx:190]:
```suggestion
                    alt='adding your tool'
```

File: pages/tools/index.page.tsx [hunk 14]
```diff
@@ -199,9 +203,12 @@ export default function ToolingPage({
...
                   rel='noreferrer'
                 >
-                  <img
+                  <Image
                     src='/img/tools/try_bowtie.png'
-                    className='rounded-sm h-[68px]'
+                    className='rounded-sm '
+                    height={68}
+                    width={100}
+                    alt={''}
```
Comment (DhairyaMajmudar) [inline pages/tools/index.page.tsx:211]:
```suggestion
                    alt='try bowtie'
```

**Review by DhairyaMajmudar — COMMENTED** [review]
@29deepanshutyagi left some suggestions to be added in the PR

File: pages/index.page.tsx (hunk 10 above)
Comment (DhairyaMajmudar) [reply pages/index.page.tsx:413]: I got you, no worries then keep the change 👍🏻

**Comment by DhairyaMajmudar** [comment]
@29deepanshutyagi there are still some comments left to be resolved, can you pls. have a look on them

File: pages/index.page.tsx
Comment (DhairyaMajmudar) [inline pages/index.page.tsx:1]: @29deepanshutyagi alt texts are missing in file pls. add them.

**Comment by 29deepanshutyagi** [comment]
kindly review it @benjagm

**Review by DhairyaMajmudar — CHANGES_REQUESTED** [review]
1. Images of case studies page is not as it was before

image

2. Sidebar Icons size is less as compare to before
 
image

3. Tooling page icons size is decreased as compared to before

image

**Comment by 29deepanshutyagi** [comment]
> @29deepanshutyagi Great progress!! just small fixes and we are good to go 🚀

but i had already set the width and height according to tailwind class there previously

**Comment by DhairyaMajmudar** [comment]
Hi @29deepanshutyagi any progress on this? We're almost there just few changes left

**Comment by 29deepanshutyagi** [comment]
> Hi @29deepanshutyagi any progress on this? We're almost there just few changes left

**Comment by heysujal** [comment]
Hi @29deepanshutyagi , I cloned your repo to have a look what the conflicts are how to fix them.

You can follow below steps to resolve these conflicts.

1. Checkout to your `check` branch.
2. Run `git remote add upstream https://github.com/json-schema-org/website.git`
3. Run `git pull upstream main`
4. Use the souce control in VSCode
5. Accept Current change for `Card.tsx`
image

Similarly you can do Accept Both option for `ToolingDetailModal.tsx`
image

After this you can stage all these merge changes(look for + icon in source control) and you can commit all the changes and push those changes like you usually do.

**Pull request closed without merging**